### ISSUE TITLE
sp.next() should not get to offsets >= 1000

### ISSUE THREAD

**Issue by mrticker** [plamere/spotipy#868]
As far as I know, the maximum number of results for any Spotify API call is 1000, so it makes no sense to go for offsets >= 1000. Still, sp.next() goes for these offsets:

```
HTTP Error for GET to https://api.spotify.com/v1/search?query=Iggy&type=playlist&offset=1000&limit=5
0 with Params: {} returned 400 due to Bad request.
```

**Comment by Peter-Schorn** [comment]
> As far as I know, the maximum number of results for any Spotify API call is 1000

This is not true. The [Get Current User's Playlists](https://developer.spotify.com/documentation/web-api/reference/#/operations/get-a-list-of-current-users-playlists) endpoint supports a maximum offset of 100,000. The [search](https://developer.spotify.com/documentation/web-api/reference/#/operations/search) endpoint supports a maximum offset of 1000. I remember prior versions of the documentation stating that offset + limit must be <= 1000. 1000 + 5 > 1000, so that's why you're getting an error response.

**Comment by mrticker** [comment]
OK. I would suggest stopping at 100000 for that method and at 1000 for the rest..

**Comment by Peter-Schorn** [comment]
Why do you believe that the other endpoints limit the offset to 1000? I don't see this in the documentation.

**Comment by zygmuntz** [comment]
I see now there are a few more with bigger offsets possible, like getting tracks for a playlist, so I believe this issue can be closed. 

Search is an example of being limited to 1000, see the offset section:
https://developer.spotify.com/documentation/web-api/reference/#/operations/search
